### ISSUE TITLE
Address caching issue

### ISSUE THREAD

**Issue by nerdstein** [d8-contrib-modules/taxonomy_menu#1]
-Updates to taxonomy -> force menu cache clear
-Add taxonomy menu -> force menu cache clear
-Update taxonomy menu -> force menu cache clear

**Comment by nerdstein** [comment]
Is there a way to target a specific cache bin?

**Comment by kylebrowning** [comment]
https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Cache%21CacheBackendInterface.php/function/CacheBackendInterface%3A%3Ainvalidate/8

**Comment by kylebrowning** [comment]
IF you want to straight up delete it, https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Cache%21CacheBackendInterface.php/function/CacheBackendInterface%3A%3Adelete/8

**Comment by nerdstein** [comment]
Awesome!

**Comment by kylebrowning** [comment]
This really is a good read. https://www.acquia.com/blog/drupal-8-performance-render-caching#animated

**Comment by nerdstein** [comment]
Per Peter W, this is not a caching issue:

Peter W
you need to implement the entity hooks
10:48
Adam Bergstein
ok, got it
then, rebuild the menu links via menu link mgr?
10:49
Peter W
NO
each time a taxonomy terms is added/updated
you tell the manager about that one term's menu link
10:49
Adam Bergstein
ah, ok
10:50
Peter W
see:  Drupal\menu_link_content\Entity::postSave()
10:50
Adam Bergstein
perfect
10:50
Peter W
you basicaly need the same kind of code in your hook
10:50
Adam Bergstein
Peter W++
thanks :)
i'll dig around
10:50
Peter W
Views module is likely doing simething similar as well on saving one view
+ doing discovery so a Views import works as expected
10:51
Adam Bergstein
cool, i'll poke around in views as well
10:51
Peter W
you are at the intersection of config + content with a config entity for a menu + taxonomy terms being content entities
10:52
Adam Bergstein
correct
10:52
Peter W
hence the need to basically handle the immediate update + discovery
10:54
Adam Bergstein
let me research and hack at this tonight a bit. @Peter W, i may come back w/ some specific qs
10:55
Peter W
ok
10:56
Peter W
Adam Bergstein:  probably you should put all the logic in the deriver class and then just call its methods from the hooks so the code is consistent
10:56
Adam Bergstein
good suggestion
10:56
Peter W
basically you need to create a definition array based on a term entity object
10:57
Adam Bergstein
i think it's already there, likely not a 1-1 to term, like you mentioned
10:57
Peter W
make a 1-1 method for term to definition as your shared code
10:57
Peter W
also that should be the target for unit testing, etc
10:58
Adam Bergstein
ill refactor it

**Comment by nerdstein** [comment]
Working on this now

**Comment by nerdstein** [comment]
Getting close on this after a frustrating weekend

**Comment by nerdstein** [comment]
Experiencing two issues that I cannot reproduce from the front-end (all seems to be working now)

```

Fail      Fatal erro Unknown              0 Unknown                            
    [05-Aug-2015 02:06:22 UTC] Uncaught PHP Exception
    Drupal\Core\Config\Schema\SchemaIncompleteException: "Schema errors for
    taxonomy_menu.taxonomy_menu.test_tax_menu with the following errors:
    taxonomy_menu.taxonomy_menu.test_tax_menu:vocabulary variable type is string
    but applied schema class is Drupal\Core\Config\Schema\Mapping,
    taxonomy_menu.taxonomy_menu.test_tax_menu:menu variable type is string but
    applied schema class is Drupal\Core\Config\Schema\Mapping" at
    /var/www/drupal/core/lib/Drupal/Core/Config/Testing/ConfigSchemaChecker.php
    line 98

```

and 

```
Fail      Browser    TaxonomyMenuOpera  102 Drupal\taxonomy_menu\Tests\Taxonomy
    I should expect to see enabled field for taxonomy term 1
Fail      Browser    TaxonomyMenuOpera  107 Drupal\taxonomy_menu\Tests\Taxonomy
    I should expect to see enabled field for taxonomy term 2
Fail      Browser    TaxonomyMenuOpera  112 Drupal\taxonomy_menu\Tests\Taxonomy
    I should expect to see enabled field for taxonomy term 3
Fail      Browser    TaxonomyMenuOpera  119 Drupal\taxonomy_menu\Tests\Taxonomy
    I should expect to see taxonomy term 2 have a parent of taxonomy term 1
```

I am pushing up a PR to stage my work and hand off.

**Comment by nerdstein** [comment]
PR: https://github.com/d8-contrib-modules/taxonomy_menu/pull/6

**Comment by nerdstein** [comment]
@EclipseGc the breaking simpletests relate to this.

**Comment by nerdstein** [comment]
This is done per last PR and help from @EclipseGc
